From: Chao Yu <chao@kernel.org>
Subject: proc: remove mark_inode_dirty() in .setattr()
Date: Tue, 31 Jan 2023 23:08:40 +0800

procfs' .setattr() has updated i_uid, i_gid and i_mode into proc dirent,
we don't need to call mark_inode_dirty() for delayed update, remove it.

Link: https://lkml.kernel.org/r/20230131150840.34726-1-chao@kernel.org
Signed-off-by: Chao Yu <chao@kernel.org>
Reviewed-by: Alexey Dobriyan <adobriyan@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/fs/proc/base.c~proc-remove-mark_inode_dirty-in-setattr
+++ a/fs/proc/base.c
@@ -700,7 +700,6 @@ int proc_setattr(struct mnt_idmap *idmap
 		return error;
 
 	setattr_copy(&nop_mnt_idmap, inode, attr);
-	mark_inode_dirty(inode);
 	return 0;
 }
 
--- a/fs/proc/generic.c~proc-remove-mark_inode_dirty-in-setattr
+++ a/fs/proc/generic.c
@@ -127,7 +127,6 @@ static int proc_notify_change(struct mnt
 		return error;
 
 	setattr_copy(&nop_mnt_idmap, inode, iattr);
-	mark_inode_dirty(inode);
 
 	proc_set_user(de, inode->i_uid, inode->i_gid);
 	de->mode = inode->i_mode;
--- a/fs/proc/proc_sysctl.c~proc-remove-mark_inode_dirty-in-setattr
+++ a/fs/proc/proc_sysctl.c
@@ -841,7 +841,6 @@ static int proc_sys_setattr(struct mnt_i
 		return error;
 
 	setattr_copy(&nop_mnt_idmap, inode, attr);
-	mark_inode_dirty(inode);
 	return 0;
 }
 
_
